jclouds-plugin/src/main/java/jenkins/plugins/jclouds/compute/JCloudsSlaveTemplate.java

```diff
@@ -312,41 +389,27 @@ public NodeMetadata get() {
         }
 
         Statement initStatement = null;
-        Statement bootstrap = null;
 
         if (this.preExistingJenkinsUser) {
             if (this.initScript.length() > 0) {
                 initStatement = Statements.exec(this.initScript);
             }
         } else {
-            // setup the jcloudTemplate to customize the nodeMetadata with jdk, etc. also opening ports
+            // provision jenkins user
             AdminAccess adminAccess = AdminAccess.builder().adminUsername(getJenkinsUser())
-                    .installAdminPrivateKey(installPrivateKey) // some VCS such as Git use SSH authentication
-                    .grantSudoToAdminUser(allowSudo) // no need
-                    .adminPrivateKey(getCloud().privateKey) // temporary due to jclouds bug
-                    .authorizeAdminPublicKey(true).adminPublicKey(getCloud().publicKey).adminHome(getFsRoot()).build();
-
+                .installAdminPrivateKey(installPrivateKey) // some VCS such as Git use SSH authentication
+                .grantSudoToAdminUser(allowSudo) // no need
+                .adminPrivateKey(getJenkinsPrivateKey()) // temporary due to jclouds bug
+                .authorizeAdminPublicKey(true).adminPublicKey(getJenkinsPublicKey()).adminHome(getFsRoot()).build();
```

Comment:
The old code looked like this:

```
AdminAccess adminAccess = AdminAccess.builder().adminUsername(getJenkinsUser())
                                        .installAdminPrivateKey(installPrivateKey) // some VCS such as Git use SSH authentication
                                        .grantSudoToAdminUser(allowSudo) // no need
                                        .adminPrivateKey(getCloud().privateKey) // temporary due to jclouds bug
                                        .authorizeAdminPublicKey(true).adminPublicKey(getCloud().publicKey).adminHome(getFsRoot()).build();
```
But this looked wrong to me (why using the *global* key in conjunction with the "jenkins" user?!)

So I changed that to use the key of the jenkins user. But that probably was a mistake. I'd really appreciate, I some jclouds developer could document AdminAccess.Builder and describe, how it's supposed to work.